### PR TITLE
added support for image preview during upload + fixed reuse banner option.

### PR DESCRIPTION
**Pull request by MrK191**

### CODE REVIEW

File: src/main/java/edu/harvard/iq/dataverse/dataverse/banners/NewBannerPage.java [hunk 1]
```diff
@@ -68,6 +71,19 @@ public String init() {
...
+    public void uploadFileEvent(FileUploadEvent event) {
+        String locale = (String) event.getComponent().getAttributes()
+                .get("imageLocale");
+
+        dto.getDataverseLocalizedBanner().stream()
+                .filter(dlb -> dlb.getLocale().equals(locale))
+                .forEach(dlb -> {
+                    dlb.setFile(event.getFile());
+                    dlb.setUploadedDisplayImage(
+                            new DefaultStreamedContent(new ByteArrayInputStream(event.getFile().getContents())));
```
Comment (madryk) [inline src/main/java/edu/harvard/iq/dataverse/dataverse/banners/NewBannerPage.java]: I think you can use different implementation of `StreamedContent` - `ByteArrayContent`. You won't need
double new operation

File: src/main/java/edu/harvard/iq/dataverse/dataverse/banners/dto/BannerMapper.java [hunk 2]
```diff
@@ -111,6 +107,29 @@ public DataverseBannerDto mapToNewBanner(Long dataverseId) {
         return dto;
     }
 
+    private DataverseLocalizedBanner MapToLocalizedBanner(DataverseBanner banner, DataverseLocalizedBannerDto fuDto) {
```
Comment (madryk) [inline src/main/java/edu/harvard/iq/dataverse/dataverse/banners/dto/BannerMapper.java]: method names should have first letter lowercased

File: src/main/java/edu/harvard/iq/dataverse/dataverse/banners/dto/DataverseLocalizedBannerDto.java [hunk 3]
```diff
@@ -32,8 +32,12 @@ public DataverseLocalizedBannerDto(Long id, String locale, byte[] image, String
 
     private StreamedContent miniDisplayImage;
 
+    private StreamedContent uploadedDisplayImage;
```
Comment (madryk) [inline src/main/java/edu/harvard/iq/dataverse/dataverse/banners/dto/DataverseLocalizedBannerDto.java]: Are you really sure that you need all of these fields in model class.
We have already 3 ones to store image.
And StreamedContent can store contentType and name.

**Review by madryk — CHANGES_REQUESTED** [review]

File: src/main/java/edu/harvard/iq/dataverse/dataverse/banners/dto/DataverseLocalizedBannerDto.java (hunk 3 above)
Comment (MrK191) [reply src/main/java/edu/harvard/iq/dataverse/dataverse/banners/dto/DataverseLocalizedBannerDto.java]: In terms of contentType and name it's true, in terms of image fields, another way would be to call some image method which would translate *byte[]* to miniature image or normal sized one.

File: src/test/java/edu/harvard/iq/dataverse/dataverse/banners/dto/BannerMapperTest.java [hunk 4]
```diff
@@ -160,7 +165,7 @@ private DataverseBannerDto createBannerDto() throws IOException {
         dataverseBannerDto.setDataverseId(1L);
 
         DataverseLocalizedBannerDto dataverseLocalizedBannerDto = new DataverseLocalizedBannerDto();
-        dataverseLocalizedBannerDto.setContentType("image/jpeg");
+        ;
```
Comment (madryk) [inline src/test/java/edu/harvard/iq/dataverse/dataverse/banners/dto/BannerMapperTest.java]: Not needed semicolon

File: src/main/java/edu/harvard/iq/dataverse/dataverse/banners/dto/BannerMapper.java [hunk 5]
```diff
@@ -109,7 +110,13 @@ private DataverseLocalizedBanner mapToLocalizedBanner(DataverseBanner banner, Da
         DataverseLocalizedBanner dataverseLocalizedBanner = new DataverseLocalizedBanner();
 
         if (isBannerReused(fuDto)) {
```
Comment (madryk) [inline src/main/java/edu/harvard/iq/dataverse/dataverse/banners/dto/BannerMapper.java]: Do we need to rely on what is in `fuDto.getFile()`? Is there is some situation when `fuDto.getDisplayedImage()` will be something different than `fuDto.getFile()`?

**Review by madryk — CHANGES_REQUESTED** [review]

**Review by madryk — APPROVED** [review]

**Pull request merged**